ide-common/src/main/java/org/digma/intellij/plugin/refreshInsightsTask/RefreshService.kt

```diff
@@ -126,12 +125,10 @@ class RefreshService(private val project: Project) {
         }
     }
 
-    private fun checkIfDataChanged(oldInsights: List<CodeObjectInsight?>, newInsights: List<CodeObjectInsight?>): Boolean {
-        val isEqual = CollectionUtils.isEqualCollection(
-                oldInsights,
-                newInsights
-        )
-        return !isEqual
+    private fun checkIfDataChanged(oldInsights: Map<String, List<CodeObjectInsight>>, newInsights: Map<String, List<CodeObjectInsight>>): Boolean {
+        // In Kotlin, you can compare two maps for equality using the == operator.
+        // The == operator checks if the two maps have the same size, the same keys, and the same values for each key.
```

Comment:
It's a well placed comment for those of who are new to kotlin

Reply:
thanks to Mykola